core/src/mender-client.c

```diff
@@ -1072,17 +1072,13 @@ mender_client_check_deployment(mender_api_deployment_data_t **deployment_data) {
         mender_log_warning("Unexpected stale deployment data");
         cJSON_Delete(mender_client_deployment_data);
     }
-    if (NULL == (mender_client_deployment_data = cJSON_CreateObject())) {
+    if ((NULL == (mender_client_deployment_data = cJSON_CreateObject()))
+        || (NULL == cJSON_AddStringToObject(mender_client_deployment_data, "id", deployment->id))
+        || (NULL == cJSON_AddStringToObject(mender_client_deployment_data, "artifact_name", deployment->artifact_name))
+        || (NULL == cJSON_AddArrayToObject(mender_client_deployment_data, "payload_types"))) {
```

Comment:
(pre-existing) Hmm, thinking about how we add/replace the `version` field -- shouldn't we do the same trick here for these fields as well? Or are they guaranteed to not be present in the deployment data?